expression/expression.go

```diff
@@ -95,13 +98,62 @@ func (triggerExpression *TriggerExpression) Evaluate() (moira.State, error) {
 	}
 }
 
-func validateUserExpression(triggerExpression *TriggerExpression, userExpression *govaluate.EvaluableExpression) (*govaluate.EvaluableExpression, error) {
-	for _, v := range userExpression.Vars() {
-		if _, err := triggerExpression.Get(v); err != nil {
-			return nil, fmt.Errorf("invalid variable value: %w", err)
+// validateUserExpression uses expr package to validate <user defined> trigger expressions
+func validateUserExpression(
+	triggerExpression *TriggerExpression,
+	userExpression *govaluate.EvaluableExpression,
+) (*govaluate.EvaluableExpression, error) {
+	expression := *triggerExpression.Expression
```

Comment:
and there is `triggerExpression.Expression` will be nil?

Reply:
This is a potentially dangerous place that could lead to panic if another developer calls this function elsewhere, it's best to protect yourself from such a thing as much as possible

Reply:
I agree, I'll push a fix now